dmutils/formats.py

```diff
@@ -76,10 +76,44 @@ def get_label_for_lot_param(lot_to_check):
     return None
 
 
+def format_field_based_price(service, question):
+    """Format a price string from a service using field mappings
+    provided in the question definition.
+
+    :param service: a service dictionary, returned from data API
+
+    :param question: dictionary describing the question containing
+                     `fields`.
+
+    :return: a formatted price string if the required
+             fields are set in the service dictionary.
+    """
+    if 'fields' in question:
+        fields = question['fields']
+        min_price_f = fields.get('minimum_price', None)
+        max_price_f = fields.get('maximum_price', None)
+        price_unit_f = fields.get('price_unit', None)
+        price_interval_f = fields.get('price_interval', None)
+
+        min_price = service.get(min_price_f, None)
```

Comment:
With `None` being the default, I think we could shorten this to
```
min_price = service.get(fields.get('minimum_price'))
```
 while still keeping it readable. The presence of the field is not really used in the price string rendering logic.